framework/src/utils/MooseUtils.C

```diff
@@ -145,6 +145,28 @@ parallelBarrierNotify(const Parallel::Communicator & comm)
   comm.barrier();
 }
 
+void serialBegin(const libMesh::Parallel::Communicator & comm)
+{
+  // unless we are the first processor...
+  if (comm.rank() > 0)
+  {
+    // ...wait for the previous processor to finish
+    int dummy;
+    comm.receive(comm.rank() - 1, dummy);
+  }
+}
+
+void serialEnd(const libMesh::Parallel::Communicator & comm)
+{
+  // unless we are the last processor...
+  if (comm.rank() + 1 < comm.size())
+  {
+    // ...notify the next processor of its turn
+    int dummy;
```

Comment:
I'm not sure if you will leave this in your code when you are done but these uninitialized values will probably create warnings in debug mode. If you do plan to leave them in you could wrap the innards of these routines with `#ifnedef NDEBUG`